discovery_engine_core/tooling/src/newscatcher.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright 2022 Xayn AG
```

Comment:
since this file is a bin as per the cargo declaration, should we put it under `bin/newscatcher.rs` instead of `src/newscatcher.rs`?

Reply:
Running `cargo new --bin` creates a `src/main.rs`, and I thought that was a bit generic, so I renamed it. I've also seen `src/bin/foo.rs` in other projects, but haven't come across just `bin/foo.rs`. No strong feelings in any direction though. The only thing I'm not sure of is if IntelliJ / VSCode plugins pick up `crate/bin/foo.rs` properly. 

Long story short: I don't particulary care, so if you do I'm happy to move it around :)

Reply:
i fully agree on the `newscatcher.rs` naming, any telling name is better than a generic `main.rs`.

regading the directory name, the [cargo book says the following](https://doc.rust-lang.org/cargo/reference/cargo-targets.html#binaries):
> Binary targets are executable programs that can be run after being compiled. The default binary filename is `src/main.rs`, which defaults to the name of the package. Additional binaries are stored in the `src/bin/` directory.

since the `newscatcher.rs` can be one of many binaries in the `tooling` crate, i'd says it's better to put it in a directory for additional binaries. i'm fine with either `src/bin/newscatcher.rs` or `bin/newscatcher` (we just have to add a `path = bin/newscatcher.rs` to the `[[bin]]` section in this case), though i have a slight preference for the latter as this would align the directory structure with the `benches/` and `examples/` directories we sometimes have.